src/services/ActivityCardService.js

```diff
@@ -1 +1,146 @@
+        await deleteFile(path);
+      }
+      await makeDirectory(path);
+
+      //Retreive all Drive files
+      await axios
+        .get(downloadUrl)
+        .then(response => {
+          driveFiles = response.data;
+        })
+        .catch(error => {
+          console.error('ERROR OCCURRED: ' + error);
+        });
+
+      //Access the Array of all files and set up path Array (to be returned)
+      const files_list = driveFiles.files;
+      var pathArr = [];
+
+      //Check upload time of each file retrived, match to current month & year
+      for (var i = 0; i < files_list.length; i++) {
+        //Read the createdTime param from Google Drive API (in ISO8601 format, so 'new Date' works here)
+        const cardUploadDate = new Date(files_list[i].createdTime);
+        const cardMonth = String(cardUploadDate.getMonth() + 1).padStart(
+          2,
+          '0',
+        );
+        const cardYear = String(cardUploadDate.getFullYear());
+
+        //download the Card and move it to the Featured Cards directory if
+        // 1. file is a .jpg (meaning it's a card)
+        // 2. Card was uploaded in the current month of the current year
+        if (
+          files_list[i].mimeType == 'image/jpeg' &&
+          cardMonth == currMonth &&
+          cardYear == currYear
+        ) {
+          await this.downloadActivityCard(files_list[i].id);
+
+          //once downloaded, check if the file exists. If it does, move into Featured Cards directory, add the path to pathArr
+          if (
+            await checkFileExists(MAINDIRECTORY + '/' + files_list[i].id + '/')
+          ) {
+            await makeDirectory(path + '/' + files_list[i].id + '/');
+            await moveFile(
+              MAINDIRECTORY + '/' + files_list[i].id + '/',
+              path + '/' + files_list[i].id + '/',
+            );
+            pathArr.push(path + '/' + files_list[i].id + '/');
+          } else {
+            throw new Error(
+              'Error occurred in downloading a Featured Activity Card.',
+            );
+          }
+        }
+      }
+
+      return pathArr;
+    } catch (e) {
+      // There was an error, catch it and do something with it
+      console.log('ERROR IN LISTING FEATURED ACTIVITY CARDS: ' + e);
+    }
+  },
+
+  /**
+   * Given the Google Drive ID of an activity card, download the card
+   * directly into local storage.
+   * @param {String} id ID of the activity card to retrieve
+   * @return {String} The requested LessonPlan object
+   */
+  downloadActivityCard: async function (id) {
+    try {
+      var path = MAINDIRECTORY + '/' + id + '/';
+      let card = {};
+
+      if (await checkFileExists(path)) {
+        throw new Error('This Activity Card is already downloaded');
+      }
+
+      const downloadUrl =
+        DRIVE_API_URLS.SEARCH_FILES +
+        id +
+        '?supportsAllDrives=true&trashed=false&includeItemsFromAllDrives=true';
+
+      await axios
```

Comment:
not sure if you need the await here since you're using the .get .then .catch promise syntax